src/plasma/plasma_store.cc

```diff
@@ -44,8 +47,24 @@ void *dlmemalign(size_t alignment, size_t bytes);
 void dlfree(void *);
 }
 
+namespace std {
+template <>
+struct hash<UniqueID> {
+  size_t operator()(const UniqueID &unique_id) const {
+    return *reinterpret_cast<const size_t *>(unique_id.id + UNIQUE_ID_SIZE -
+                                             sizeof(size_t));
```

Comment:
This is using the last `size_t` bytes of the object ID as a hash of the unique ID? Why not the following?

```
return *reinterpret_cast<const size_t *>(unique_id.id)
```